crates/rspack_util/src/defer.rs

```diff
@@ -0,0 +1,29 @@
+use std::mem::ManuallyDrop;
+
+pub struct DeferGuard(ManuallyDrop<Box<dyn FnOnce()>>);
```

Comment:
Box doesn't seem necessary?

Reply:
![image](https://github.com/user-attachments/assets/a82965cd-5e23-4ff8-8b55-3d49e3adf212)

`FnOnce` should use `ManuallyDrop::take` move ownership.

Reply:
we can keep the generics here

Reply:
see https://docs.rs/scopeguard/latest/scopeguard/